Realm/RLMArrayLinkView.mm

```diff
@@ -148,7 +151,10 @@ - (void)insertObject:(RLMObject *)object atIndex:(NSUInteger)index {
     RLMLinkViewArrayValidateInWriteTransaction(self);
 
     RLMValidateObjectClass(object, self.objectClassName);
-    if (object.realm != self.realm) {
+    if (index > _backingLinkView->size()) {
+        @throw RLMException(@"Trying to insert object at invalid index");
```

Comment:
this message could be better -- `NSArray` says `Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArrayM insertObject:atIndex:]: index 200 beyond bounds [0 .. 0]'`